### PR TITLE
Adds new analytics events for notifications

### PR DESCRIPTION
**Pull request by roundhill**
Adds new events for notifications settings, and also fixes some incomplete events related to push authentication.

### CODE REVIEW

**Comment by roundhill** [comment]
cc @jleandroperez can you have a peek at this to make sure iOS will match up?

**Comment by jleandroperez** [comment]
@roundhill everything looks perfect!. iOS is blocked until [this pr gets reviewed](https://github.com/wordpress-mobile/WordPressCom-Analytics-iOS/pull/29), but those are exactly the strings i've got around here.

Thanks for checking Dan!

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java [hunk 1]
```diff
@@ -103,6 +103,10 @@
         PUSH_AUTHENTICATION_FAILED,
         PUSH_AUTHENTICATION_IGNORED,
         SETTINGS_LANGUAGE_SELECTION_FORCED,
+        NOTIFICATION_SETTINGS_LIST_OPENED,
+        NOTIFICATION_SETTINGS_STREAMS_OPENED,
```
Comment (daniloercoli) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java]: Where does these events are going to be used?
I'm even more curious about `NOTIFICATION_SETTINGS_STREAMS_OPENED` :)

**Comment by daniloercoli** [comment]
Aside from the question above :shipit:

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java (hunk 1 above)
Comment (roundhill) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java]: Oh, I should have clarified. I'm going to use these in https://github.com/wordpress-mobile/WordPress-Android/issues/2525

**Pull request merged**